incubator/hnc/pkg/controllers/hierarchy_controller.go

```diff
@@ -208,24 +208,29 @@ func (r *HierarchyReconciler) markExisting(log logr.Logger, ns *forest.Namespace
 	}
 }
 
+// syncRequiredChildOf propagates the required child value from the forest to the spec if possible
+// (the spec itself will be synced next), or removes the requiredChildOf value if there's a problem
+// and notifies the would-be parent namespace.
 func (r *HierarchyReconciler) syncRequiredChildOf(log logr.Logger, inst *api.HierarchyConfiguration, ns *forest.Namespace) {
 	if ns.RequiredChildOf == "" {
 		return
 	}
 
-	switch {
-	case inst.Spec.Parent == "":
+	switch inst.Spec.Parent {
+	case "":
 		log.Info("Required subnamespace: initializing", "parent", ns.RequiredChildOf)
 		inst.Spec.Parent = ns.RequiredChildOf
-	case inst.Spec.Parent == ns.RequiredChildOf:
+	case ns.RequiredChildOf:
 		// ok
 	default:
-		log.Info("Required subnamespace: assigned to wrong parent", "intended", ns.RequiredChildOf, "actual", inst.Spec.Parent)
-		r.enqueueAffected(log, "incorrect parent of the subnamespace", inst.Spec.Parent)
-		msg := fmt.Sprintf("required child of %s but parent is set to %s", ns.RequiredChildOf, inst.Spec.Parent)
-		r.enqueueAffected(log, "wrong parent set as a parent", inst.Spec.Parent)
```

Comment:
The updated code in green on the right side lgtm, but I got confused why we had `enqueueAffected inst.Spec.Parent` twice here?

Reply:
Um. Almost certainly a bug 😄